[14.0][MIG] sale_timesheet

MiquelRForgeFlow opened the pull request:
Supersedes https://github.com/OCA/OpenUpgrade/pull/2878.

Murtaza-OSI committed:
[14.0][MIG] sale_timesheet

MiquelRForgeFlow committed:
[MIG] sale_timesheet (continuation)

JulianaPoudou commented:
Hello @MiquelRForgeFlow ,
I launched the migration with the last commit of your script, the error below occurred:

2021-12-20 14:36:35,814 83074 ERROR fil14 OpenUpgrade: Error after 0:00:00.000710 running 
        UPDATE project_project
        SET allow_billable = TRUE
        WHERE pricing_type IS NOT NULL 
2021-12-20 14:36:35,814 83074 ERROR fil14 OpenUpgrade: sale_timesheet: error in migration script /OpenUpgrade/openupgrade_scripts/scripts/sale_timesheet/14.0.1.0/post-migration.py: new row for relation "project_project" violates check constraint "project_project_timesheet_product_required_if_billable_and_time"

Going back to the previous version of this script in the function fill_allow_billable, the script is passed now (update timesheet_product_id)

def fill_allow_billable(env):
    timesheet_product_id = env.ref("sale_timesheet.time_product", False)
    openupgrade.logged_query(
        env.cr,
        """
        UPDATE project_project
        SET allow_billable = TRUE,
            timesheet_product_id = %s
        WHERE pricing_type IS NOT NULL""",
        (timesheet_product_id.id,),
    )

So, my question is why did you delete this part in this function? 
Thanks and regards, 
Juliana

pedrobaeza committed:
[OU-FIX+IMP] sale_timesheet: Preserve value + timesheet product

- FIX: Fill properly project timesheet product.
- IMP: Preserve old billable_type values: Any other module may need it.

pedrobaeza reviewed (approved):
Added an improvement for preserving old billable_type values for any other module requiring it, and populate properly the timesheet product in projects.
